modules/oe_theme_helper/src/Plugin/field_group/FieldGroupFormatter/FieldInPageNavigationFormatter.php

```diff
@@ -0,0 +1,188 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_theme_helper\Plugin\field_group\FieldGroupFormatter;
+
+use Drupal\field_group\FieldGroupFormatterBase;
+
+/**
+ * Format a field group to create in-page navigation.
+ *
+ * @FieldGroupFormatter(
+ *   id = "oe_theme_helper_in_page_navigation",
+ *   label = @Translation("Field in-page navigation"),
+ *   description = @Translation("Format field group for ecl-inpage-navigation."),
+ *   supported_contexts = {
+ *     "view"
+ *   }
+ * )
+ */
+class FieldInPageNavigationFormatter extends FieldGroupFormatterBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function process(&$element, $processed_object) {
+    $element += [
+      '#type' => 'field_group_in_page_navigation',
+    ];
+    // Get children elements from the in-page navigation group.
+    $field_elements = $processed_object['#fieldgroups'][$element['#group_name']]->children;
+    $links = [];
+    foreach ($field_elements as $value) {
+      // Check if we should add the field to page contents.
+      if (!is_array($element[$value]) || $this->excludeField($value, $element[$value], $processed_object['#fieldgroups']) === TRUE) {
+        continue;
+      }
+      // We will add anchors when the element, field or group, has a label.
+      $label = $this->fieldGetLabel($value, $processed_object);
+      if ($label === '') {
+        continue;
+      }
+      $type = $this->groupOrField($value, $processed_object['#fieldgroups']);
+      // If a group has a default id, use it as anchor link.
+      if ($type === 'group' && !empty($processed_object['#fieldgroups'][$value]->format_settings['id'])) {
+        $id = $processed_object['#fieldgroups'][$value]->format_settings['id'];
+      }
+      else {
+        $id = "inline-nav-" . $i = isset($i) ? ++$i : 1;
+      }
+
+      $links[] = [
+        'href' => "#" . $id,
+        'label' => $label,
+      ];
+      $element[$value]['#id'] = $id;
+      $element[$value]['#attributes']['id'] = $id;
+    }
+
+    $element['#links'] = $links;
+    if (count($links) > 0) {
+      $element['#inline_title'] = t('Page contents');
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function preRender(&$element, $rendering_object) {
+    parent::preRender($element, $rendering_object);
+    $this->process($element, $rendering_object);
+  }
+
+  /**
+   * Check if we should exclude this field from the in-page navigation menu.
+   *
+   * @param string $field_name
+   *   Name of the field.
+   * @param array $field
+   *   The field to check.
+   * @param array $field_groups
+   *   List of groups.
+   *
+   * @return bool
+   *   Exclude or not the field.
+   */
+  protected function excludeField(string $field_name, array $field, array $field_groups) :bool {
```

Comment:
Remove this method as we will check for emptiness in the template.

Reply:
Removed